hw/ip/otbn/doc/_index.md

```diff
@@ -361,6 +361,72 @@ Both use the same state for tracking control flow.
 This is a stack of tuples containing a loop count, start address and end address.
 The stack has a maximum depth of eight and the top of the stack is the current loop.
 
+### Loop nesting
+
+OTBN permits loop nesting and branches and jumps inside loops.
+However, it doesn't have support for early termination of loops: there's no way to pop an entry from the loop stack without executing the last instruction of the loop the correct number of times.
+It can also only pop one level of the loop stack per instruction.
+
+To avoid polluting the loop stack or avoid surprising behaviour, the programmer must ensure that:
+* Even if there are branches and jumps within a loop body, the final instruction of the loop body gets executed exactly once per iteration.
+* Nested loops have distinct end addresses.
+* The end instruction of an outer loop is not executed before an inner loop finishes.
```

Comment:
Strictly you can do this, if you eventually jump back to a point where the inner loop end instruction gets executed. However it feels like an odd thing to do so felt it better to warn against doing it and added the 'surprising behaviour' as this won't inherently pollute the loop stack but probably indicates you've messed up something so it will eventually pollute the loop stack.